CMakeLists.txt

```diff
@@ -27,41 +33,102 @@ option(
     ${PROJECT_IS_TOP_LEVEL}
 )
 
+# includes
 include(GNUInstallDirs)
+include(CMakePackageConfigHelpers)
 
-add_library(beman.inplace_vector INTERFACE)
+add_library(beman_inplace_vector INTERFACE)
 # [CMAKE.LIBRARY_ALIAS]
-add_library(beman::inplace_vector ALIAS beman.inplace_vector)
+add_library(beman::beman_inplace_vector ALIAS beman_inplace_vector)
 
-target_include_directories(
-    beman.inplace_vector
+target_sources(
+    beman_inplace_vector
+    PUBLIC
+        FILE_SET inplace_vector_public_headers
+        TYPE HEADERS
+        BASE_DIRS ${CMAKE_CURRENT_SOURCE_DIR}/include
+        FILES
+            "${CMAKE_CURRENT_SOURCE_DIR}/include/beman/inplace_vector/inplace_vector.hpp"
+)
+set_target_properties(
+    beman_inplace_vector
+    PROPERTIES VERIFY_INTERFACE_HEADER_SETS ON
+)
+target_compile_features(
+    beman_inplace_vector
     INTERFACE
-        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
-        $<INSTALL_INTERFACE:include>
+        "$<$<COMPILE_FEATURES:cxx_std_23>:cxx_std_23>"
+        "$<$<NOT:$<COMPILE_FEATURES:cxx_std_23>>:cxx_std_20>"
 )
 
-# Install the InplaceVector library to the appropriate destination
-install(
-    TARGETS beman.inplace_vector
-    EXPORT ${TARGETS_EXPORT_NAME}
-    DESTINATION
-    ${CMAKE_INSTALL_LIBDIR}
-)
+# export cmake config package
+block()
+    # NOTE: copied from execution26:
+    # FIXME: but not yet used? CK
+    # set(TARGET_NAME inplace_vector)
+    # set(TARGET_NAMESPACE beman)
+    # set(TARGET_PREFIX ${TARGET_NAMESPACE}.${TARGET_NAME})
+    # set(TARGET_LIBRARY ${PROJECT_NAME})
+    # set(TARGET_ALIAS ${TARGET_NAMESPACE}::${TARGET_LIBRARY})
 
-# Install the header files to the appropriate destination
-install(
-    DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/
-    DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/${CMAKE_PROJECT_NAME}
-    FILES_MATCHING
-    PATTERN
-    "${CMAKE_CURRENT_SOURCE_DIR}/include/beman/inplace_vector/inplace_vector.hpp"
-)
+    set(TARGET_PACKAGE_NAME ${PROJECT_NAME}-config)
+    set(TARGETS_EXPORT_NAME ${PROJECT_NAME}-targets)
+    set(INSTALL_CONFIGDIR ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME})
+
+    # Install the InplaceVector library to the appropriate destination
```

Comment:
Need help reviewing this

Reply:
Yes, most people do this **wrong** or only **partly** ....